src/frontend/ts/state/chessStore.ts

```diff
@@ -131,18 +119,30 @@ const useChessStore = create<ChessState>((set, get) => ({
       case Update.Result: {
         const resultData = data as ResultUpdate
         const gameID = resultData.gameID
+        const currentGame = get().activeGames.get(gameID)
+        const archivedGame = get().archivedGames.get(gameID)
 
-        const displayGame = get().displayGame
-        if ((displayGame !== null) && (gameID === displayGame.info.gameID)) {
-          get().setDisplayGame(null)
+        // copy moves to archived version before deleting
+        const updatedGame: ArchivedGameInfo = {
+          ...archivedGame,
+          moves: currentGame.moves
         }
 
+        //  save the archive + moves in state
+        set(state => ({ archivedGames: state.archivedGames.set(gameID, updatedGame) }))
```

Comment:
nit: what you have is correct, though we could join these two `set` calls into one call on line `137`:
```
set(state => ({
  activeGames: activeGames,
  archivedGames: state.archivedGames.set(gameID, updatedGame)
}))
```